### PR TITLE
#6654: Moving init for self.compute_kernel_config

### PR DESCRIPTION
**Pull request by eyonland**

### CODE REVIEW

**Review by cfjchu — APPROVED** [review]

**Review by xanderchin — APPROVED** [review]

**Review by arakhmati — APPROVED** [review]

**Comment by tt-rkim** [comment]
@TT-billteng forcing this through since @eyonland has indicated this is high priority work and cannot wait on @boris-drazic 's approval.

**Pull request merged**
